From bba42ca72233679b98a5c38d6d6aefb4c2d78b60 Mon Sep 17 00:00:00 2001
From: Carlos Perez Lopez <carlospzlz@gmail.com>
Date: Mon, 22 May 2017 23:02:26 +0100
Subject: [PATCH] Addressed issue 992.

- Fixed dangling quotes produced by cmake file.
- Added token stringification to convert OPENSUBDIV_VERSION_STRING into
a string.
---
 CMakeLists.txt                               | 2 +-
 examples/glEvalLimit/glEvalLimit.cpp         | 4 +++-
 examples/glFVarViewer/glFVarViewer.cpp       | 4 +++-
 examples/glImaging/glImaging.cpp             | 6 ++++--
 examples/glPaintTest/glPaintTest.cpp         | 4 +++-
 examples/glPtexViewer/glPtexViewer.cpp       | 4 +++-
 examples/glShareTopology/glShareTopology.cpp | 4 +++-
 examples/glStencilViewer/glStencilViewer.cpp | 4 +++-
 examples/glViewer/glViewer.cpp               | 4 +++-
 9 files changed, 26 insertions(+), 10 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e02ce90c..6d4f5342 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -48,7 +48,7 @@ endif()
     string(REGEX REPLACE "^v" "" OSD_SONAME ${OSD_SONAME})
 
     add_definitions(
-        -DOPENSUBDIV_VERSION_STRING="${OSD_SONAME}"
+        -DOPENSUBDIV_VERSION_STRING=${OSD_SONAME}
     )
 
 #-------------------------------------------------------------------------------
diff --git a/examples/glEvalLimit/glEvalLimit.cpp b/examples/glEvalLimit/glEvalLimit.cpp
index 2786de7c..789155f3 100644
--- a/examples/glEvalLimit/glEvalLimit.cpp
+++ b/examples/glEvalLimit/glEvalLimit.cpp
@@ -22,6 +22,8 @@
 //   language governing permissions and limitations under the Apache License.
 //
 
+#define STRINGIFY(a) #a
+
 #include "../common/glUtils.h"
 
 #include <GLFW/glfw3.h>
@@ -1460,7 +1462,7 @@ int main(int argc, char **argv) {
         return 1;
     }
 
-    static const char windowTitle[] = "OpenSubdiv glEvalLimit " OPENSUBDIV_VERSION_STRING;
+    static const char windowTitle[] = "OpenSubdiv glEvalLimit " STRINGIFY(OPENSUBDIV_VERSION_STRING);
 
     GLUtils::SetMinimumGLVersion();
 
diff --git a/examples/glFVarViewer/glFVarViewer.cpp b/examples/glFVarViewer/glFVarViewer.cpp
index 615aba51..72143f2b 100644
--- a/examples/glFVarViewer/glFVarViewer.cpp
+++ b/examples/glFVarViewer/glFVarViewer.cpp
@@ -22,6 +22,8 @@
 //   language governing permissions and limitations under the Apache License.
 //
 
+#define STRINGIFY(a) #a
+
 #include "../common/glUtils.h"
 
 #include <GLFW/glfw3.h>
@@ -1230,7 +1232,7 @@ int main(int argc, char ** argv) {
         return 1;
     }
 
-    static const char windowTitle[] = "OpenSubdiv glFVarViewer " OPENSUBDIV_VERSION_STRING;
+    static const char windowTitle[] = "OpenSubdiv glFVarViewer " STRINGIFY(OPENSUBDIV_VERSION_STRING);
 
     GLUtils::SetMinimumGLVersion();
 
diff --git a/examples/glImaging/glImaging.cpp b/examples/glImaging/glImaging.cpp
index ac4b275d..0be4c8f8 100644
--- a/examples/glImaging/glImaging.cpp
+++ b/examples/glImaging/glImaging.cpp
@@ -22,6 +22,8 @@
 //   language governing permissions and limitations under the Apache License.
 //
 
+#define STRINGIFY(a) #a
+
 #include "../common/glUtils.h"
 
 #include <iostream>
@@ -529,7 +531,7 @@ int main(int argc, char ** argv) {
     }
 
     static const char windowTitle[] =
-        "OpenSubdiv imaging test " OPENSUBDIV_VERSION_STRING;
+        "OpenSubdiv imaging test " STRINGIFY(OPENSUBDIV_VERSION_STRING);
 
     GLUtils::SetMinimumGLVersion();
 
@@ -616,7 +618,7 @@ int main(int argc, char ** argv) {
         ofs << "<body>\n";
         ofs << "<h3>OpenSubdiv imaging regression test<h3>\n";
         ofs << "<pre>\n";
-        ofs << "OpenSubdiv      : " << OPENSUBDIV_VERSION_STRING << "\n";
+        ofs << "OpenSubdiv      : " << STRINGIFY(OPENSUBDIV_VERSION_STRING) << "\n";
         ofs << "GL Version      : " << glGetString(GL_VERSION)
             << ", " << glGetString(GL_VENDOR)
             << ", " << glGetString(GL_RENDERER)
diff --git a/examples/glPaintTest/glPaintTest.cpp b/examples/glPaintTest/glPaintTest.cpp
index c5688055..0470a3af 100644
--- a/examples/glPaintTest/glPaintTest.cpp
+++ b/examples/glPaintTest/glPaintTest.cpp
@@ -23,6 +23,8 @@
 //
 
 
+#define STRINGIFY(a) #a
+
 #include "../common/glUtils.h"
 
 #include <GLFW/glfw3.h>
@@ -1105,7 +1107,7 @@ int main(int argc, char ** argv) {
         return 1;
     }
 
-    static const char windowTitle[] = "OpenSubdiv glPaintTest " OPENSUBDIV_VERSION_STRING;
+    static const char windowTitle[] = "OpenSubdiv glPaintTest " STRINGIFY(OPENSUBDIV_VERSION_STRING);
 
     GLUtils::SetMinimumGLVersion();
 
diff --git a/examples/glPtexViewer/glPtexViewer.cpp b/examples/glPtexViewer/glPtexViewer.cpp
index d8950f22..e370b955 100644
--- a/examples/glPtexViewer/glPtexViewer.cpp
+++ b/examples/glPtexViewer/glPtexViewer.cpp
@@ -22,6 +22,8 @@
 //   language governing permissions and limitations under the Apache License.
 //
 
+#define STRINGIFY(a) #a
+
 #if defined(__APPLE__)
     #if defined(OSD_USES_GLEW)
         #include <GL/glew.h>
@@ -1783,7 +1785,7 @@ int main(int argc, char ** argv) {
         return 1;
     }
 
-    static const char windowTitle[] = "OpenSubdiv glPtexViewer" OPENSUBDIV_VERSION_STRING;
+    static const char windowTitle[] = "OpenSubdiv glPtexViewer" STRINGIFY(OPENSUBDIV_VERSION_STRING);
 
     GLUtils::SetMinimumGLVersion();
 
diff --git a/examples/glShareTopology/glShareTopology.cpp b/examples/glShareTopology/glShareTopology.cpp
index 71c90fdf..32d68805 100644
--- a/examples/glShareTopology/glShareTopology.cpp
+++ b/examples/glShareTopology/glShareTopology.cpp
@@ -22,6 +22,8 @@
 //   language governing permissions and limitations under the Apache License.
 //
 
+#define STRINGIFY(a) #a
+
 #include "../common/glUtils.h"
 
 #include <GLFW/glfw3.h>
@@ -1180,7 +1182,7 @@ int main(int argc, char ** argv) {
         return 1;
     }
 
-    static const char windowTitle[] = "OpenSubdiv batching example " OPENSUBDIV_VERSION_STRING;
+    static const char windowTitle[] = "OpenSubdiv batching example " STRINGIFY(OPENSUBDIV_VERSION_STRING);
 
     GLUtils::SetMinimumGLVersion();
 
diff --git a/examples/glStencilViewer/glStencilViewer.cpp b/examples/glStencilViewer/glStencilViewer.cpp
index 48a8f7d6..01878788 100644
--- a/examples/glStencilViewer/glStencilViewer.cpp
+++ b/examples/glStencilViewer/glStencilViewer.cpp
@@ -22,6 +22,8 @@
 //   language governing permissions and limitations under the Apache License.
 //
 
+#define STRINGIFY(a) #a
+
 #include "../common/glUtils.h"
 
 #include <GLFW/glfw3.h>
@@ -1077,7 +1079,7 @@ int main(int argc, char **argv) {
         return 1;
     }
 
-    static const char windowTitle[] = "OpenSubdiv glStencilViewer " OPENSUBDIV_VERSION_STRING;
+    static const char windowTitle[] = "OpenSubdiv glStencilViewer " STRINGIFY(OPENSUBDIV_VERSION_STRING);
 
     GLUtils::SetMinimumGLVersion();
 
diff --git a/examples/glViewer/glViewer.cpp b/examples/glViewer/glViewer.cpp
index 14f6e218..5a2ef2f7 100644
--- a/examples/glViewer/glViewer.cpp
+++ b/examples/glViewer/glViewer.cpp
@@ -22,6 +22,8 @@
 //   language governing permissions and limitations under the Apache License.
 //
 
+#define STRINGIFY(a) #a
+
 #include "../common/glUtils.h"
 
 #include <GLFW/glfw3.h>
@@ -1682,7 +1684,7 @@ int main(int argc, char ** argv) {
         return 1;
     }
 
-    static const char windowTitle[] = "OpenSubdiv glViewer " OPENSUBDIV_VERSION_STRING;
+    static const char windowTitle[] = "OpenSubdiv glViewer " STRINGIFY(OPENSUBDIV_VERSION_STRING);
 
     GLUtils::SetMinimumGLVersion(argc, argv);
 
